### DISCUSSION TITLE
Callback ID generation with crypto.getRandomValues()

### DISCUSSION THREAD

**Discussion by sabljak** [josefarias/hotwire_combobox#223] [Ideas]
Hotwire Combobox uses `crypto.randomUUID` to [generate random Callback ID](https://github.com/search?q=repo:josefarias/hotwire_combobox%20crypto&type=code) but that seems to be the problem when running the app in an HTTP environment since `randomUUID()` is available only in secure contexts.

What do you think about generating the Callback ID some other way, a custom implementation with crypto.getRandomValues() since it is the only member of the Crypto interface that can be used from an insecure context.

I can submit a PR if you are interested.

👋

**Comment by josefarias** [comment]
Thanks for bringing that up @sabljak

This callback id doesn't need to be cryptographically secure. We can go with something simpler than `crypto.getRandomValues()` if needed.

Curious, is the combobox not working for you because of this? Or is it just generating console warnings with no effect?

Also curious, what's the use-case for not running on HTTPS? Is this an internal app or something?

Not opposed to making the change, just want to understand the reasons for it before we do it.

**Reply by sabljak** [reply]
Yes, the combobox is not working because of this, it is throwing errors in the console.

<img width="371" alt="Screenshot 2024-10-14 at 15 08 47" src="https://github.com/user-attachments/assets/eb434729-2fa5-41b2-a754-1e6d4c92ac68">

The use-case for not running the app on HTTPS is a review app environment.

**Reply by josefarias** [reply]
Got it. Yeah, I'd welcome a PR to use any random string that's practically guaranteed not to have collisions.

`crypto.getRandomValues()` is fine to use as long as everything happens in a short and simple function.

**Reply by sabljak** [reply]
Great!

I will create a PR addressing this issue soon.

Thanks! 🙌

**Comment by sabljak** [comment]
Starting to work on this 🤞

**Comment by josefarias** [comment]
Merged! Thank you for this!

https://github.com/josefarias/hotwire_combobox/pull/233
